src/core_init_atmosphere/mpas_init_atm_gwd.F

```diff
@@ -7,15 +7,35 @@
 !
 module mpas_init_atm_gwd
 
+   use iso_c_binding, only : c_char, c_int, c_float, c_ptr, c_loc
+
    use mpas_derived_types, only : MPAS_LOG_ERR
    use mpas_framework
    use mpas_timekeeping
    use mpas_log, only : mpas_log_write
+   use mpas_c_interfacing, only : mpas_f_to_c_string
 
    public :: compute_gwd_fields
 
    private
 
+   interface
+      subroutine read_geogrid(fname, rarray, nx, ny, nz, &
+                             & isigned, endian, scalefactor, wordsize, status) bind(C)
+         use iso_c_binding, only : c_char, c_int, c_float, c_ptr
+         character (c_char), intent(in) :: fname
```

Comment:
Depending on how we ultimately need to declare `c_filename`, we may need to adjust the declaration of `fname` here.